crates/starknet-rs-py/src/types/transactions/declare.rs

```diff
@@ -0,0 +1,94 @@
+use crate::{
+    cached_state::PyCachedState,
+    types::{
+        general_config::PyStarknetGeneralConfig,
+        transaction_execution_info::PyTransactionExecutionInfo,
+    },
+};
+use cairo_felt::Felt252;
+use num_bigint::BigUint;
+use pyo3::{exceptions::PyValueError, prelude::*};
+use starknet_rs::{
+    business_logic::{
+        fact_state::in_memory_state_reader::InMemoryStateReader, state::cached_state::CachedState,
+        transaction::objects::internal_declare::InternalDeclare,
+    },
+    definitions::{
+        constants::VALIDATE_DECLARE_ENTRY_POINT_SELECTOR, transaction_type::TransactionType,
+    },
+    services::api::contract_class::ContractClass,
+    utils::{Address, ClassHash},
+};
+
+#[pyclass(subclass)]
+#[pyo3(name = "InternalDeclare")]
+pub struct PyInternalDeclare {
+    inner: InternalDeclare,
+}
+
+#[pymethods]
+impl PyInternalDeclare {
+    #[new]
+    fn new(
+        hash_value: BigUint,
+        version: u64,
+        max_fee: u64,
+        signature: Vec<BigUint>,
+        nonce: BigUint,
+        class_hash: ClassHash,
+        sender_address: BigUint,
+    ) -> Self {
+        let sender_address = Address(Felt252::from(sender_address));
+        let signature = signature.into_iter().map(Felt252::from).collect();
+        let nonce = Felt252::from(nonce);
+        let hash_value = Felt252::from(hash_value);
+        let contract_class =
```

Comment:
Just a question, is OK to set this contract class to a default one?
Is there any reason why we are not using the InternalDeclare::new?

Reply:
The [devnet](https://github.com/Shard-Labs/starknet-devnet/blob/e54507eff2ffd7945b380262361997ed24c61663/starknet_devnet/transactions.py#L282-L289) only uses it for creating empty `InternalDeclare`s. In the future, all constructors should use pyo3's signature macro attribute -- like `PyStarknetGeneralConfig`'s constructor -- to mimic cairo-lang's interface. In that case this would be:
```rust
#[pyo3(signature = (
    ...
    contract_class = ContractClass::new(Default::default(), Default::default(), None).unwrap(),
    ...
))]
```